### PR TITLE
Youtube amp

### PR DESCRIPTION
**Pull request by annez**
This adds `<amp-youtube>` functionality.

### CODE REVIEW

**Comment by AileenCGN** [comment]
Hey @annez 

Thanks for the PR and sorry for the delay! I will have a look at this in a bit...

**Comment by ponkys** [comment]
Hi @annez  and @AileenCGN , It would be amazing if could land this pr. I came to a moment that this could be really useful. Thank you very much for your time and effort! 👍

**Comment by jbhannah** [comment]
@annez 👋 Thanks! Can you rebase onto `master` please? I'll try to get this merged.

**Comment by annez** [comment]
Hey @jbhannah I've amended the entire thing. I also added `request_timeout` config override, as certain requests can fail but only due to a slow network and we wanted to cater for that.

A few things that are probably worth considering for future iterations (and stuff that we do):

1. Do we want to fallback to invalid AMP HTML when X thing fails? The example here is `<amp-img>` - we're taking the conservative approach of doing nothing, but in a build system where it's taking stuff from X and making static files, I would like it to ensure that it at least spits out valid AMP? I'd rather it fixed the markup for valid AMP than otherwise. Maybe a config option such as `force` that would then force the markup to be valid?

**Comment by kevinansfield** [comment]
@annez is this commit needed https://github.com/jbhannah/amperize/pull/113/commits/6e2ff880d0a403c979b28c21f545a51272d98924?

Image `src` attributes can be http and shouldn't be auto changed to https, amp downloads+caches images and then serves them via https, if the url gets changed http->https by `amperize` then it can result in 404s. At the moment the tests verify this incidentally, we probably should have a specific test for that but it would be good to do that separately to this youtube implementation.

**Comment by annez** [comment]
@kevinansfield We were experiencing issues with this but will revert in accordance to spec :)

File: test/amperize.test.js [hunk 1]
```diff
@@ -390,7 +425,7 @@ describe('Amperize', function () {
         });
 
         it('adds \'https\' protocol to <iframe> if only \'http\' protocol is supplied', function (done) {
-            var url = '<iframe src="http://giphy.com/embed/3oEduKP4VaUxJvLwuA" width="480" height="372" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="http://giphy.com/gifs/afv-funny-fail-lol-3oEduKP4VaUxJvLwuA">via GIPHY</a></p></p>';
+            var url = '<iframe src="https://giphy.com/embed/3oEduKP4VaUxJvLwuA" width="480" height="372" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="http://giphy.com/gifs/afv-funny-fail-lol-3oEduKP4VaUxJvLwuA">via GIPHY</a></p></p>';
```
Comment (kevinansfield) [inline test/amperize.test.js]: It looks like this change means the test isn't testing what it says it is?

```suggestion
            var url = '<iframe src="http://giphy.com/embed/3oEduKP4VaUxJvLwuA" width="480" height="372" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="http://giphy.com/gifs/afv-funny-fail-lol-3oEduKP4VaUxJvLwuA">via GIPHY</a></p>';
```

**Comment by annez** [comment]
@kevinansfield Updated accordingly :)

**Comment by kevinansfield** [comment]
@annez this has just been released in 0.6.0 🎉

**Pull request merged**